File: JvmClient/src/commonMain/kotlin/spaceEngineers/model/ToolbarItem.kt
package spaceEngineers.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable


interface ToolbarItem {
    val name: String
    val enabled: Boolean
}

@Serializable
data class DataToolbarItem(
    @SerialName("Name")
    override val name: String,
    @SerialName("Enabled")
    override val enabled: Boolean
) : ToolbarItem

@Serializable
data class DataToolbarItemActions(
    @SerialName("ActionId")
    val actionId: String,
    @SerialName("Name")
    override val name: String,
    @SerialName("Enabled")
    override val enabled: Boolean
) : ToolbarItem

@Serializable
data class DataToolbarItemTerminalBlock(
    @SerialName("BlockEntityId")
    val blockEntityId: Long,
    @SerialName("ActionId")
    val actionId: String,
    @SerialName("Name")
    override val name: String,
    @SerialName("Enabled")
    override val enabled: Boolean
) : ToolbarItem

@Serializable
data class DataToolbarItemDefinition(
    @SerialName("Id")
    val id: DefinitionId,
    @SerialName("Name")
    override val name: String,
    @SerialName("Enabled")
    override val enabled: Boolean
) : ToolbarItem {

    override fun toString(): String {
        return "$id - $name"
    }
}

File: Source/Ivxr.SpaceEngineers/WorldModel/ToolbarItem.cs
namespace Iv4xr.SpaceEngineers.WorldModel
{
    public class ToolbarItem
    {
        public string Name;
        public bool Enabled;
    }
    
    public class ToolbarItemDefinition: ToolbarItem
    {
        public DefinitionId Id;
    }


    public class ToolbarItemActions : ToolbarItem
    {
        public string ActionId;
    }
    
    public class ToolbarItemTerminalBlock : ToolbarItemActions
    {
        public long BlockEntityId;
    }
}
